src/ray/raylet/node_manager.cc

```diff
@@ -2723,6 +2723,20 @@ void NodeManager::TriggerGlobalGC() {
   should_local_gc_ = true;
 }
 
+void NodeManager::HandleGetDriverAndWorkerPids(
+    rpc::GetDriverAndWorkerPidsRequest request,
+    rpc::GetDriverAndWorkerPidsReply *reply,
+    rpc::SendReplyCallback send_reply_callback) {
+  auto all_workers = worker_pool_.GetAllRegisteredWorkers(/* filter_dead_worker */ true);
```

Comment:
by default this will include the "IO workers" that are used for spilling objects. I don't think we want to include them (at least for now). You can pass `filter_io_workers=true` to filter them.